packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -82,50 +119,124 @@ const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
   }
 };
 
-const saveExecution = async (cumulusMessage, knex) => {
-  const executionModel = new Execution();
-  const executionArn = getMessageExecutionArn(cumulusMessage);
-
-  const isRDSWriteEnabled = await shouldWriteExecutionToRDS(cumulusMessage, knex);
+const saveExecutionViaTransaction = async ({ cumulusMessage, trx }) =>
+  trx(tableNames.executions)
+    .insert({
+      arn: getMessageExecutionArn(cumulusMessage),
+      cumulus_version: getMessageCumulusVersion(cumulusMessage),
+    });
 
-  if (!isRDSWriteEnabled) {
+const saveExecution = async ({
+  cumulusMessage,
+  knex,
+  executionModel = new Execution(),
+}) =>
+  knex.transaction(async (trx) => {
+    await saveExecutionViaTransaction({ cumulusMessage, trx });
     return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
-  }
+  });
 
-  try {
-    return await knex.transaction(async (trx) => {
-      await trx(tableNames.executions)
-        .insert({
-          arn: executionArn,
-          cumulus_version: getMessageCumulusVersion(cumulusMessage),
-        });
-      return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+const savePdrViaTransaction = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  trx,
+}) =>
+  trx(tableNames.pdrs)
+    .insert({
+      name: getMessagePdrName(cumulusMessage),
+      status: getWorkflowStatus(cumulusMessage),
+      collectionCumulusId: collection.cumulusId,
+      providerCumulusId: provider.cumulusId,
     });
-  } catch (error) {
-    log.error(`Failed to write execution records for ${executionArn}`, error);
-    throw error;
+
+const savePdr = async ({
```

Comment:
I moved to destructuring an options object partially because I think it's cleaner when passing lots of options and also because I wanted pass in the model as an argument, which makes test mocking much easier